cmd/spiffe-helper/main.go

```diff
@@ -13,29 +15,45 @@ func main() {
 	// 1. Create Sidecar
 	// 2. Run Sidecar's Daemon
 
+	log := logger.Std
 	configFile := flag.String("config", "helper.conf", "<configFile> Configuration file path")
 	flag.Parse()
 
 	config, err := ParseConfig(*configFile)
 	if err != nil {
-		log.Fatalf("error parsing configuration file: %v\n%v", *configFile, err)
+		log.Errorf("error parsing configuration file: %v\n%v", *configFile, err)
+		panic(err)
+	}
+	timeout, err := GetTimeout(config)
+	if err != nil {
+		log.Errorf("error parsing timeout: %s\n%v", config.Timeout, err)
+		panic(err)
 	}
+	config.Log = log
 
-	log.Printf("Connecting to agent at %q\n", config.AgentAddress)
+	log.Infof("Connecting to agent at %q\n", config.AgentAddress)
 	if config.Cmd == "" {
-		log.Println("Warning: no cmd defined to execute.")
+		log.Warnf("No cmd defined to execute.")
 	}
 
-	log.Printf("Using configuration file: %q\n", *configFile)
+	log.Infof("Using configuration file: %q\n", *configFile)
 
-	spiffeSidecar, err := sidecar.NewSidecar(config)
+	spiffeSidecar := sidecar.NewSidecar(config)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	err = spiffeSidecar.RunDaemon(ctx)
 	if err != nil {
 		panic(err)
 	}
 
-	ctx := context.Background()
-	err = spiffeSidecar.RunDaemon(ctx)
-	if err != nil {
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, os.Interrupt)
+	select {
+	case err = <-spiffeSidecar.ErrChan:
```

Comment:
Instead of exporting this channel, why can't we make RunDaemon blocking, move the error channel here, kick off the goroutine here in main, and simply send down the error channel with the result of RunDaemon? Exposing an exported error channel here feels a little clunky. One of the things patterns we try to avoid  is the "start/stop" pattern (this was a motivator to rewrite go-spiffe v1), instead preferring blocking functions and leaving it up to callers to decide if they need to spawn a goroutine to call it. Sometimes they do, sometimes they don't. Start/stop forces callers into a specific concurrency pattern.